Bio/Seq.py

```diff
@@ -1259,6 +1260,47 @@ def ungap(self, gap=None):
             raise ValueError("Unexpected gap character, {0!r}".format(gap))
         return Seq(str(self).replace(gap, ""), alpha)
 
+    def join(self, other):
+        """Return a concatination of all sequences in the iterable 'other', spaced with the sequence from self.
+
+        Accepts Seq objects and Strings as objects to be concatinated with the spacer
+
+        >>> spacer = Seq('NNNNN')
+        >>> seqlist = list([Seq("AAA"),Seq("TTT"),Seq("PPP")])
+        >>> concatenated = spacer.join(seqlist)
+        >>> concatenated
+        Seq('AAANNNNNTTTNNNNNPPP')
+
+        Throws error if other is not an iterable and if objects inside of the iterable are not Seq or String objects
+        """
+        if not isinstance(other, collections.Iterable):
+            raise ValueError("Input must be an iterable")
+        from Bio.SeqRecord import SeqRecord  # Lazy to avoid circular imports
+        temp_data = ""
+        if self._data == "":  # if the spacer is empty it will initially default to the type of the first sequence
+            a = None
```

Comment:
What if the arguments are all alphabet-less strings? Add ``if a is None: a = self.alphabet`` logic at the end?